tests/PHPStanTests/nsrt/preg-match.php

```diff
@@ -27,6 +27,13 @@ function doMatch(string $s): void
         assertType('array{}', $matches);
     }
     assertType('array{}|array{0: string, 1?: string|null}', $matches);
+
+    if (Preg::isMatch('/Price: (?P<currency>£|€)\d+/', $s, $matches)) {
```

Comment:
something is wrong with Hoa parsing the AST for the pattern.

it works as expected with a pattern like

```suggestion
    if (Preg::isMatch('/Price: (£|€)\d+/', $s, $matches)) {
```